### PR TITLE
GH-42208: [Java] Fix the Test in flight-sql-jdbc-driver Module

### PR DESCRIPTION
**Pull request by llama90**
### Rationale for this change

Adjust assertion message order for JUnit 5 compatibility.

### What changes are included in this PR?

- Reordering the arguments of `assertEquals(expected, actual, message)`

### Are these changes tested?

Yes.

### Are there any user-facing changes?

No.

* GitHub Issue: #42208

### CODE REVIEW

**Comment by llama90** [comment]
@vibhatha Could you execute `@github-actions crossbow submit java-jars`?

I found the reason, it was the wrong order of arguments in `assertEquals`. However, I couldn't figure out why it occurs intermittently at that time.

File: java/flight/flight-sql-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ITDriverJarValidation.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ private static JarFile getJdbcJarFile() throws IOException {
 
     assertNotNull(driverClassURL, "Driver jar was not detected in the classpath");
     assertEquals(
-        "Driver jar was not detected in the classpath", "jar", driverClassURL.getProtocol());
+        "jar", driverClassURL.getProtocol(), "Driver jar was not detected in the classpath");
```
Comment (vibhatha) [inline java/flight/flight-sql-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ITDriverJarValidation.java:73]: Both are strings, so the IDE won't complaint in the first place, but how did we miss this in the CIs?

**Review by lidavidm — APPROVED** [review]

**Comment by lidavidm** [comment]
```
Error:    TestAceroSubstraitConsumer.testRunExtendedExpressionsProjectAndFilter:637 NullPointer Cannot invoke "org.apache.arrow.vector.ValueIterableVector.getValueIterable()" because "nameVector" is null
```

might be another thing to fix

File: java/flight/flight-sql-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ITDriverJarValidation.java (hunk 1 above)
Comment (llama90) [reply java/flight/flight-sql-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ITDriverJarValidation.java:73]: Also, I'm curious about that too. 

Actually, I have executed multiple `mvn clean install` commands to verify the changes. I never saw that message before merging the PR (to deal with the Flight Module) into the main branch. 🥲

Perhaps, I suspect that this workflow might be cached, similar to how the JNI workflow failed.

Comment (vibhatha) [reply java/flight/flight-sql-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ITDriverJarValidation.java:73]: Probably... It is fine, let's move forward. Thanks for quickly working on this fix.

**Review by vibhatha — APPROVED** [review]

**Comment by llama90** [comment]
> ```
> Error:    TestAceroSubstraitConsumer.testRunExtendedExpressionsProjectAndFilter:637 NullPointer Cannot invoke "org.apache.arrow.vector.ValueIterableVector.getValueIterable()" because "nameVector" is null
> ```
> 
> might be another thing to fix

I  checked the error, and it is caused by this PR. Can I open an issue and resolve it?

- https://github.com/apache/arrow/pull/41895

**Pull request merged**